amps/ags/rm-community/rm-community-repo/source/java/org/alfresco/module/org_alfresco_module_rm/disposition/DispositionServiceImpl.java

```diff
@@ -477,12 +477,9 @@ public NodeRef getAssociatedRecordsManagementContainer(DispositionSchedule dispo
                 {
                     // TODO in the future we should be able to support disposition schedule reuse, but for now just warn that
                     //      only the first disposition schedule will be considered
-                    if (LOGGER.isWarnEnabled())
-                    {
-                        LOGGER.warn("Retention schedule has more than one associated records management container.  " +
-                                "This is not currently supported so only the first container will be considered. " +
-                                "(dispositionScheduleNodeRef=" + dispositionSchedule.getNodeRef().toString() + ")");
-                    }
+                    LOGGER.atWarn().log("Retention schedule has more than one associated records management container.  " +
```

Comment:
Is there a reason why we replaced the traditional `LOGGER.warn(message, arguments...)` with the fluent-logging variant? As far as the documentation states, they're perfectly equivalent: https://www.slf4j.org/manual.html#fluent

![image](https://github.com/Alfresco/alfresco-community-repo/assets/24280982/8b09f741-c542-4c87-92a7-3b0221f5dc12)

Reply:
In the "Hello world" example they are equivalent, but in the code we have a String concatenation which we only want to happen if the logging level is at WARN or above. There are a few ways to achieve this. In the old code we had an if statement which disrupts the flow of the code. In the traditional slf4j we can use:
```
LOGGER.warn("NodeRef {}", schedule.getNodeRef());
```
However this does incur a (tiny) overhead in calling `getNodeRef()` which causes PMD to complain. Alternatively we can use the fluent syntax of:
```
LOGGER.atWarn().log("NodeRef {}", schedule.getNodeRef());
```
(or even simply concatenating the strings is fine from a performance pov)

I can't imagine many customers are going to set their log level lower than WARN, so the whole example is a bit pointless, but the AGS code has hundreds of debug examples that could be converted.